WordPress/Classes/ViewRelated/Reader/ReaderCardsStreamViewController.swift

```diff
@@ -3,20 +3,32 @@ import Foundation
 class ReaderCardsStreamViewController: ReaderStreamViewController {
     private var currentPage = 1
 
+    private let readerCardTopicsIdentifier = "ReaderTopicsCell"
+
     lazy var cardsService: ReaderCardService = {
         return ReaderCardService()
     }()
 
     override func viewDidLoad() {
         super.viewDidLoad()
         ReaderWelcomeBanner.displayIfNeeded(in: tableView)
+        tableView.register(ReaderSuggestedTopicsCell.self, forCellReuseIdentifier: readerCardTopicsIdentifier)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        // Only sync if the tableview is at the top, otherwise this will change offset
+        if isTableViewAtTheTop() {
+            super.viewWillAppear(animated)
+        }
     }
 
     // MARK: - TableView Related
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         if let posts = content.content as? [ReaderCard], let cardPost = posts[indexPath.row].post {
             return cell(for: cardPost, at: indexPath)
+        } else if let posts = content.content as? [ReaderCard], let interests = posts[indexPath.row].interests?.array as? [ReaderTagTopic] {
```

Comment:
So something doesn't feel right in this function. 
1. First, take the data source out:
Something like : 
```
 lazy var readerCards: [ReaderCard]? = {
        return content.content as? [ReaderCard]
    }()
```
And then look at the specific card:
 `let readerCard = readerCards[indexPath.row]`
I also recommend implementing:
```
 func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return readerCards?.count ?? 0
       }
```
So you'll have more confidence that you're not out of bound when calling cell for row.

2. We want to make this scaleable for other types of cards, right ? Why don't we save the type of card from the remote and switch on that?

Reply:
Yeah, I agree with you.

I don't like this implementation, I'll try to make it better.

Reply:
Ok! I simplified that a lot! Let me know what do you think. :)

Reply:
So much better!!! 👏